### PR TITLE
xocl: Supporting passthrough functionality when userPf in VM and mgmt is in Host

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Supporting passthrough functionality when userPf in VM and mgmt is in Host

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c [hunk 1]
```diff
@@ -848,34 +848,35 @@ void xclmgmt_mailbox_srv(void *arg, void *data, size_t len,
...
-					(void) xocl_peer_response(lro, req->req, msgid, &ret,
+			p2p_bar_addr = mb_p2p->p2p_bar_addr;
+			p2p_bar_len = mb_p2p->p2p_bar_len;
+			mgmt_info(lro, "got the p2p bar addr = %lld\n", p2p_bar_addr);
+			mgmt_info(lro, "got the p2p bar len = %lld\n", p2p_bar_len);
+			if (!p2p_bar_addr) {
+				pci_write_config_byte(pdev, 0x188, 0x0);
+				pci_write_config_dword(pdev, 0x190, 0x0);
+				pci_write_config_dword(pdev, 0x194, 0x0);
+				pci_write_config_dword(pdev, 0x18C, 0x0);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c]: can we add some comments and change numbers like 0x188 to some meaningful macro names here? the current code is not readable and will be broken for sure in the future, if we keep it this way.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 2]
```diff
@@ -397,6 +407,9 @@ int xocl_hot_reset(struct xocl_dev *xdev, u32 flag)
 		xocl_reset_notify(xdev->core.pdev, false);
 
 		xocl_drvinst_set_offline(xdev->core.drm, false);
+
+		//Reset Virtualization registers
+		(void) xocl_mb_read_p2p_addr(xdev);
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c]: Why does it have to refresh bridge address table after resetting FPGA?

Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c]: Need to configure these PCIe extended registers again

Comment (maxzhen) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c]: The question is why you need to configure it again? We do not reset the on-device pcie bridge, so why it lost the configuration?

**Pull request merged**